source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -394,6 +394,7 @@ void DnsCacheImpl::finishResolve(const std::string& host,
     primary_host_info->host_info_->setAddresses(new_address, std::move(address_list));
 
     runAddUpdateCallbacks(host, primary_host_info->host_info_);
+    primary_host_info->host_info_->firstResolveComplete();
```

Comment:
Do you mean : primary_host_info->host_info_->setFirstResolveComplete() here?

Reply:
Yes, your are right. correct it in latest change

Reply:
Good to see the CI is now passed.
One more question: this PR moved first_resolve_complete_ to true to a few lines below to cover runAddUpdateCallbacks().  Just a thought: do we need to move it even further to below line 410 to cover notifyAllThreads() and runResolutionCompleteCallBacks() to avoid other potential issues?

Reply:
Based on my understanding as of now, notifyAllThreads will invoke the pending request (depends on the host resolve here) decoding. If do this further change the first_resolve_complete_ is still false when notifyAllThreads is called.

Reply:
Yeah, there is no need to move this further late as there is existing logic to set it right after this block, here:

https://github.com/envoyproxy/envoy/blob/fcf17c9161dc7c396e2a9edfa8d138a4ed19249c/source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc#L409

LGTM for the code change.